docfx_yaml/extension.py

```diff
@@ -1547,6 +1553,78 @@ def search_cross_references(obj, current_object_name: str, known_uids: List[str]
                     markdown_utils.reformat_markdown_to_html(attribute_type))
 
 
+# Type alias used for toc_yaml entries.
+_toc_yaml_type_alias = dict[str, any]
+
+def merge_markdown_and_package_toc(
+    app: sphinx.application,
+    pkg_toc_yaml: list[_toc_yaml_type_alias],
+    markdown_toc_yaml: _toc_yaml_type_alias,
+    known_uids: set[str],
+) -> tuple[MutableSet[str], list[_toc_yaml_type_alias]]:
+    """
+    Merges the markdown and package table of contents.
+
+    Args:
+        app: Sphinx application.
+        pkg_toc_yaml: table of content for package files.
+        markdown_toc_yaml: table fo content for markdown files.
+
+    Returns:
+        A set of markdown pages that has been added, and the merged table of
+        contents file, with files in the correct position.
+    """
+    def _flatten_toc(
+        toc_yaml_entry: list[_toc_yaml_type_alias],
+    ) -> list[_toc_yaml_type_alias]:
+        """Flattens and retrieves all children within pkg_toc_yaml."""
+        toc_queue = list(toc_yaml_entry)
+        for entry in toc_queue:
+            if (children := entry.get('items')):
+                toc_queue.extend(_flatten_toc(children))
+                toc_queue.extend(children)
```

Comment:
This is still modifying `toc_queue` while looping over it. Can you iterate over `toc_yaml_entry` instead?

If I'm understanding correctly, we don't need a queue at all. Just need to iterate over the "graph" and put it all into a single list.

Reply:
Oops 🤦 Made updates.